src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -243,49 +237,42 @@ private bool StartCore(ProcessStartInfo startInfo)
                 }
             }
 
-            int childPid = -1, stdinFd = -1, stdoutFd = -1, stderrFd = -1, result = -1;
+            int childPid, stdinFd, stdoutFd, stderrFd;
             string[] envp = CreateEnvp(startInfo);
             string cwd = !string.IsNullOrWhiteSpace(startInfo.WorkingDirectory) ? startInfo.WorkingDirectory : null;
 
-            filename = ResolvePath(startInfo.FileName); 
-            if (!string.IsNullOrEmpty(filename))
+            if (!startInfo.UseShellExecute)
```

Comment:
nit, not worth changing here -- usually if there's two equal codepaths, I would put the positive case first, as it's easier to read